File: jdroid-android-core/src/main/java/com/jdroid/android/shortcuts/analytics/FirebaseAppShortcutsAnalyticsTracker.java
package com.jdroid.android.shortcuts.analytics;

import android.os.Bundle;

import com.google.firebase.analytics.FirebaseAnalytics;
import com.jdroid.android.firebase.analytics.AbstractFirebaseAnalyticsTracker;

public class FirebaseAppShortcutsAnalyticsTracker extends AbstractFirebaseAnalyticsTracker implements AppShortcutsAnalyticsTracker {
	
	@Override
	public void trackPinShortcut(String shortcutName) {
		Bundle params = new Bundle();
		params.putString(FirebaseAnalytics.Param.ITEM_ID, shortcutName);
		getFirebaseAnalyticsHelper().sendEvent("pin_shortcut", params);
	}
}

File: jdroid-android-core/src/main/java/com/jdroid/android/shortcuts/analytics/FirebaseAppShortcutsAnalyticsTracker.kt
package com.jdroid.android.shortcuts.analytics

import com.google.firebase.analytics.FirebaseAnalytics
import com.jdroid.android.firebase.analytics.AbstractFirebaseAnalyticsTracker
import com.jdroid.android.firebase.analytics.FirebaseAnalyticsParams

class FirebaseAppShortcutsAnalyticsTracker : AbstractFirebaseAnalyticsTracker(), AppShortcutsAnalyticsTracker {

    override fun trackPinShortcut(shortcutName: String) {
        val params = FirebaseAnalyticsParams()
        params.put(FirebaseAnalytics.Param.ITEM_ID, shortcutName)
        getFirebaseAnalyticsHelper().sendEvent("pin_shortcut", params)
    }
}
